### ISSUE TITLE
About character-aware model

### ISSUE THREAD

**Issue by helson73** [OpenNMT/OpenNMT#24]
Hi, thanks for releasing such a beautiful toolkit !
This library seems like follow https://github.com/harvardnlp/seq2seq-attn,
however compare to seq2seq-attn, cnn-based character-aware encoder is not available.
Is there any plan to add such feature(cnn) in the future?

**Comment by srush** [comment]
We don't have plans to do this in the short-term, as it complicates the input and output formatting. An alternative approach is to use subword encodings like BPE which can be wrapped around the whole system. We will release tools to do this in the near future, but in the mean time there are several that exist online (https://github.com/vteromero/byte-pair-encoding)

**Comment by srush** [comment]
Actually looks like BPE is coming very soon: in this PR https://github.com/OpenNMT/OpenNMT/pull/27

**Comment by helson73** [comment]
@srush Thanks, unfortunately in my language pair, applying BPE is not possible. 
I just implemented it myself, and also advanced Bayesian dropouts,
`https://arxiv.org/abs/1512.05287`, and still tuning. 
I am wondering why don't you guys add this version of dropout? Current dropout is too naive.
I found it can be implemented in a very simple way. Just rewrite nn.Dropout to force noise be fixed during all time steps, and tie noise tensors between clones, and also edit LSTM.lua to apply dropout to all nodes which feed to nn.Linear Nodes. I will share the result when the exp is complete.
